EIPS/eip-5018.md

```diff
@@ -0,0 +1,125 @@
+---
+eip: 5018
+title: Directory Standard
+description: A standard interface for filesystem directories.
+author: Qi Zhou (@qizhou)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-04-18
+---
+
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for filesystem directories within smart contracts.
+This standard provides basic functionality to read/write binary objects for any size, as well as allow reading/writing chunks of the object if the object is too large to fit in a single transaction.
+
+## Motivation
+
+A standard interface allows any binary objects on EVM-based blockchain to be re-used by other dApps.
```

Comment:
Would you mind going a bit more in depth on why this EIP is necessary? Is storing a tree of binary data on-chain really that common?

Reply:
Thanks for the comment.  I have added the motivation with the applications with EIP-4804.  With the standard, we can put web contents (e.g., HTML/SVG) in the same tree structure as the local directory and reference them using protocol-relative URLs, i.e, web3:// defined by EIP-4804.

Please let me know if the motivation is clear or not.

Reply:
Just uploaded the homepages of ENS and Uniswap on Rinkeby using the directory standard and EIP-4804:

- ENS:
https://galileo.web3q.io/ensdomains.eth:4/
or 
https://galileo.web3q.io/0x9e081Df45E0D167636DB9C61C7ce719A58d82E3b:4/

https://galileo.web3q.io/uniswaps.eth:4/
or
https://galileo.web3q.io/0xC100d49e8F3d621E4438E82E4f95CF505b3E2a28:4/